### PR TITLE
[WIP] Add local file support

### PR DESCRIPTION
**Pull request by charjac**

### CODE REVIEW

**Comment by charjac** [comment]
I add a simple proove of concept, i had to use the http api to serve the file.

File: app/actions/local.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    payload
+  };
+}
+
+export function scanLocalFoldersSuccess(payload) {
+  return {
+    type: SCAN_LOCAL_FOLDER_SUCCESS,
+    payload: payload.reduce((acc, track) => ({
+      ...acc,
+      [track.id]: track
```
[the remark this answers is not in the document]
Comment (charjac) [reply app/actions/local.js]: It start here server/http/api/localfile.js, then it goes there server/file/index.js, then an event is send to the IpcContainer and the action is dispatched into redux.
The id is a simple uuid/v4.

[the remark this answers is not in the document]
Comment (nukeop) [reply app/actions/local.js]: Nice, this is a good way to do this

**Comment by charjac** [comment]
I think it only need some ui. How do you imagine it ?

File: server/http/api/localfile.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          .filter(artist => artist.includes(req.body.artist))
+          .map(artist => byArtist[artist])
+          .flat()
+          .filter(track => track.name.includes(req.body.track))
+          .map(track => ({
+            id: track.id,
+            title: track.name,
+            duration: track.duration,
+            source: 'Local',
+            stream: `http://127.0.0.1:${port}/nuclear/file/${track.id}`,
```
Comment (nukeop) [inline server/http/api/localfile.js]: Maybe it's possible to use the file:// scheme?

File: app/containers/SoundContainer/index.js [hunk 3]
```diff
@@ -161,6 +161,8 @@ class SoundContainer extends React.Component {
       ).stream;
     }
 
+    console.log('render sound', streamUrl, player.seek);
```
Comment (nukeop) [inline app/containers/SoundContainer/index.js]: debug statement?

**Comment by nukeop** [comment]
Very good. The way I see it is a list of directories that you've already added to your library, with simple controls like add more directories, remove those already added, etc. Maybe scan them right away, or put a button for scanning in there too.

For now, it'll be fine if we just have a long list of TrackRow components under that, representing all songs in your library. Later we can think about browsing by album, artist, etc. Maybe make the list sortable, using something like this: https://react.semantic-ui.com/collections/table/

File: server/http/api/localfile.js (hunk 2 above)
Comment (charjac) [reply server/http/api/localfile.js]: if we want to use file:// schema we will have to disable some electron security (dont remember everything but the guy on stackoverflow said it is not a good idea).

File: app/containers/SoundContainer/index.js (hunk 3 above)
Comment (charjac) [reply app/containers/SoundContainer/index.js]: oops :)

**Comment by nukeop** [comment]
I'm going to checkout the code tomorrow and do some more thorough review of the functionality as it's a huge feature and I want to do it justice.

**Comment by nukeop** [comment]
I noticed that if there's no album art the queue item popup is broken. It'll be worthwhile to fix that after this is merged.

**Comment by nukeop** [comment]
I'd put the menu where you add folders in the Library section, since it's empty initially and there's no obvious way of adding music. Probably at the top, or in a tab.

**Comment by nukeop** [comment]
This time I think it'll be crucial to have good options for browsing the local library, so we'll need sorting, filtering (with an input box at the top), and maybe album view like some players.

**Comment by charjac** [comment]
i will work on this today

**Comment by charjac** [comment]
it works, but its not beautiful :D.

**Comment by nukeop** [comment]
Awesome, this is really a great feature. We can think about making it beautiful over time, what matters is that it's implemented and it works.

**Comment by nukeop** [comment]
What this means is that all features for 0.5.0 are here and we can release soon!

**Pull request merged**